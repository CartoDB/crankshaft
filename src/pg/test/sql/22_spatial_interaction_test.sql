SET client_min_messages TO WARNING;
\set ECHO none
\pset format unaligned

\i test/fixtures/spatial_interactions.sql

-- Gravity Model
\echo '--> gravity model'
SELECT
  a.cartodb_id,
  spint.predicted,
  a.flow_data,
  round((spint.coeffs->>'origin_i')::numeric, 4) As coeff_origin_i,
  round((spint.coeffs->>'destination_j')::numeric, 4) As coeff_destination_j,
  round((spint.coeffs->>'intercept')::numeric, 4) As coeff_intercept,
  round(spint.r_squared, 4) As r_squared,
  round(spint.aic, 4) as aic
FROM cdb_crankshaft.CDB_SpIntGravity('select * from austria_test',
  'flow_data',
  Array['origin_i'],
  Array['destination_j'],
  'dij') As spint
JOIN austria_test As a
on a.cartodb_id = spint.rowid
ORDER BY a.cartodb_id ASC
LIMIT 2;


-- Production-constrained
\echo '--> production-constrained'
SELECT
  a.cartodb_id,
  spint.predicted,
  a.flow_data,
  round((spint.coeffs->>'origin_4016')::numeric, 4) As coeff_origin_4016,
  round((spint.coeffs->>'destination_j')::numeric, 4) As coeff_destination_j,
  round((spint.coeffs->>'intercept')::numeric, 4) As coeff_intercept,
  round(spint.r_squared, 4) As r_squared,
  round(spint.aic, 4) as aic
FROM cdb_crankshaft.CDB_SpIntProduction('select * from austria_test',
  'flow_data',
  'origin_i',
  Array['destination_j'],
  'dij') As spint
JOIN austria_test As a
on a.cartodb_id = spint.rowid
ORDER BY a.cartodb_id ASC
LIMIT 2;


-- Attraction-constrained
\echo '--> attraction-constrained'
SELECT
  a.cartodb_id,
  -- spint.coeffs::text,
  spint.predicted,
  a.flow_data,
  round((spint.coeffs->>'dest_5790')::numeric, 4) As coeff_dest_at34,
  round((spint.coeffs->>'destination_j')::numeric, 4) As coeff_destination_j,
  round((spint.coeffs->>'intercept')::numeric, 4) As coeff_intercept,
  round((spint.coeffs->>'dij')::numeric, 4) As coeff_cost,
  round(spint.r_squared, 4) As r_squared,
  round(spint.aic, 4) as aic
FROM cdb_crankshaft.CDB_SpIntAttraction('select * from austria_test',
  'flow_data',
  'origin_i',
  Array['destination_j'],
  'dij') As spint
JOIN austria_test As a
on a.cartodb_id = spint.rowid
ORDER BY a.cartodb_id ASC
LIMIT 2;


-- Doubly-constrained
\echo '--> doubly-constrained'
SELECT
  a.cartodb_id,
  -- spint.coeffs::text,
  spint.predicted,
  a.flow_data,
  round((spint.coeffs->>'dest_25741')::numeric, 4) As coeff_dest_25741,
  round((spint.coeffs->>'origin_4897')::numeric, 4) As coeff_origin_4897,
  round((spint.coeffs->>'intercept')::numeric, 4) As coeff_intercept,
  round((spint.coeffs->>'dij')::numeric, 4) As coeff_cost,
  round(spint.r_squared, 4) As r_squared,
  round(spint.aic, 4) as aic
FROM cdb_crankshaft.CDB_SpIntDoubly('select * from austria_test',
  'flow_data',
  'origin_i',
  'destination_j',
  'dij') As spint
JOIN austria_test As a
on a.cartodb_id = spint.rowid
ORDER BY a.cartodb_id ASC
LIMIT 2;


-- Local Production-constrained
\echo '--> local gravity'
SELECT
  a.cartodb_id,
  a.flow_data,
  round((spint.coeffs->>'origin_i')::numeric, 4) As coeff_origin_i,
  round((spint.coeffs->>'destination_j')::numeric, 4) As coeff_destination_j,
  round((spint.coeffs->>'dij')::numeric, 4) As coeff_cost,
  round(spint.r_squared, 4) As r_squared,
  round(spint.aic, 4) as aic
FROM cdb_crankshaft.CDB_SpIntLocalGravity('select * from austria_test',
  'flow_data',
  Array['origin_i'],
  Array['destination_j'],
  'origin',
  'dij') As spint
JOIN austria_test As a
on a.cartodb_id = spint.rowid
ORDER BY a.cartodb_id ASC
LIMIT 10;


-- Local Production-constrained
\echo '--> local production-constrained'
SELECT
  a.cartodb_id,
  a.flow_data,
  round((spint.coeffs->>'destination_j')::numeric, 4) As coeff_destination_j,
  round((spint.coeffs->>'dij')::numeric, 4) As coeff_cost,
  round(spint.r_squared, 4) As r_squared,
  round(spint.aic, 4) as aic
FROM cdb_crankshaft.CDB_SpIntLocalProduction('select * from austria_test',
  'flow_data',
  'origin',
  Array['destination_j'],
  'dij') As spint
JOIN austria_test As a
on a.cartodb_id = spint.rowid
ORDER BY a.cartodb_id ASC
LIMIT 10;

-- Local Production-constrained
\echo '--> local attraction-constrained'
SELECT
  a.cartodb_id,
  a.flow_data,
  round((spint.coeffs->>'origin_i')::numeric, 4) As coeff_origin_i,
  round((spint.coeffs->>'dij')::numeric, 4) As coeff_cost,
  round(spint.r_squared, 4) As r_squared,
  round(spint.aic, 4) as aic
FROM cdb_crankshaft.CDB_SpIntLocalAttraction('select * from austria_test',
  'flow_data',
  'destination',
  Array['origin_i'],
  'dij') As spint
JOIN austria_test As a
on a.cartodb_id = spint.rowid
ORDER BY a.cartodb_id ASC
LIMIT 10;
