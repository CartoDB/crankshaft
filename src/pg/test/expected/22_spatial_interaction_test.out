SET client_min_messages TO WARNING;
\set ECHO none
--> gravity model
cartodb_id|predicted|flow_data|coeff_origin_i|coeff_destination_j|coeff_intercept|r_squared|aic
1|0.2328542752225343|0|1.2181|1.1938|-15.1410|0.8077|37923.4744
2|1255.5399375668803|1131|1.2181|1.1938|-15.1410|0.8077|37923.4744
(2 rows)
--> production-constrained
cartodb_id|predicted|flow_data|coeff_origin_4016|coeff_destination_j|coeff_intercept|r_squared|aic
1|0.23786759806324095|0|0.6543|1.2002|-5.6827|0.8092|37639.5714
2|1346.6209890796022|1131|0.6543|1.2002|-5.6827|0.8092|37639.5714
(2 rows)
--> attraction-constrained
cartodb_id|predicted|flow_data|coeff_dest_at34|coeff_destination_j|coeff_intercept|coeff_cost|r_squared|aic
1|0.23786759806324095|0|1.0131|1.2002|-5.6827|0.0096|0.8092|37639.5714
2|1346.6209890796022|1131|1.0131|1.2002|-5.6827|0.0096|0.8092|37639.5714
(2 rows)
--> doubly-constrained
cartodb_id|predicted|flow_data|coeff_dest_25741|coeff_origin_4897|coeff_intercept|coeff_cost|r_squared|aic
1|0.22669249250364376|0|2.8708|0.8350|3.5304|0.0097|0.8124|37030.2911
2|1208.493396391691|1131|2.8708|0.8350|3.5304|0.0097|0.8124|37030.2911
(2 rows)
--> local gravity
cartodb_id|flow_data|coeff_origin_i|coeff_destination_j|coeff_cost|r_squared|aic
1|0|0.4123|0.2632|0.0083|0.3203|1268.1738
2|1131|0.5461|0.1855|0.0101|0.5577|901.7820
3|1887|0.3019|0.4335|0.0082|0.3589|3893.9736
4|69|0.0733|0.7093|0.0098|0.6974|2960.1637
5|738|0.1519|0.6109|0.0109|0.7539|1500.6016
6|98|0.0964|0.6078|0.0085|0.4715|2767.9888
7|31|-0.7697|1.7466|0.0131|0.9665|2009.0465
8|43|-0.8679|1.7729|0.0138|0.9696|1341.4094
9|19|-1.0010|1.5351|0.0086|0.8671|976.6408
(9 rows)
--> local production-constrained
cartodb_id|flow_data|coeff_destination_j|coeff_cost|r_squared|aic
1|0|0.2632|0.0083|0.3203|1268.1738
2|1131|0.1855|0.0101|0.5577|901.7820
3|1887|0.4335|0.0082|0.3589|3893.9736
4|69|0.7093|0.0098|0.6974|2960.1637
5|738|0.6109|0.0109|0.7539|1500.6016
6|98|0.6078|0.0085|0.4715|2767.9888
7|31|1.7466|0.0131|0.9665|2009.0465
8|43|1.7729|0.0138|0.9696|1341.4094
9|19|1.5351|0.0086|0.8671|976.6408
(9 rows)
--> local attraction-constrained
cartodb_id|flow_data|coeff_origin_i|coeff_cost|r_squared|aic
1|0|0.2207|0.0081|0.3290|1003.2007
2|1131|0.1168|0.0093|0.4617|1121.5678
3|1887|0.4877|0.0079|0.3552|3504.3820
4|69|0.6732|0.0093|0.6241|2766.4621
5|738|0.4116|0.0101|0.7789|873.7556
6|98|0.5479|0.0089|0.5229|1626.3788
7|31|1.5728|0.0128|0.9552|1547.9533
8|43|2.0717|0.0128|0.9745|1737.1971
9|19|1.6484|0.0081|0.8967|918.4579
(9 rows)
